src/main/java/com/faforever/client/map/MapService.java

```diff
@@ -303,25 +303,20 @@ public Image loadPreview(String mapName, PreviewSize previewSize) {
 
 
   public ObservableList<MapBean> getInstalledMaps() {
-    return installedSkirmishMaps;
+    return installedMaps;
   }
 
-
   public Optional<MapBean> getMapLocallyFromName(String mapFolderName) {
     logger.debug("Trying to find map '{}' locally", mapFolderName);
-    ObservableList<MapBean> installedMaps = getInstalledMaps();
-    synchronized (installedMaps) {
-      for (MapBean mapBean : installedMaps) {
-        if (mapFolderName.equalsIgnoreCase(mapBean.getFolderName())) {
-          logger.debug("Found map {} locally", mapFolderName);
-          return Optional.of(mapBean);
-        }
-      }
+    String mapFolderKey = mapFolderName.toLowerCase();
+    if (!mapsByFolderName.containsKey(mapFolderKey)) {
+      return Optional.empty();
+    } else {
+      MapBean mapBean = mapsByFolderName.get(mapFolderKey);
+      return Optional.of(mapBean);
```

Comment:
I'd either remove the else, therefore having it as an abort condition or swap if and else around.